File: DSA/Aarohi-Sharma/Day1_q2.cpp
#include<iostream>
using namespace std;
int main(){
    int n;
    cin>>n;
    int array[n];
    cout<<"Enter elements:";
    for (int i=0; i<n; i++){
        cin>>array[i];
    }
    for (int i=0; i<n; i++){
        cout<<array[i]<<endl;
    }
    int k;
    cin>>k;
    int count=0;
    for(int i=0; i<n; i++){
        for (int j=i+1; j<n; j++){
            if ((i+j)%k==0){
                count++;
            }
        }
    }
    cout<<count;
}

File: DSA/Aanidhay-Aggarwal/Day-1/Day1_q2.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class CountPairs {
    public static int countPairs(int[] arr, int k) {
        Map<Integer, Integer> count = new HashMap<>();
        int pairs = 0;

        for (int num : arr) {
            int remainder = num % k;
            pairs += count.getOrDefault((k - remainder) % k, 0);
            count.put(remainder, count.getOrDefault(remainder, 0) + 1);
        }

        return pairs;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the size of the array: ");
        int n = scanner.nextInt();

        int[] arr = new int[n];
        System.out.print("Enter the array elements (space-separated): ");
        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }

        System.out.print("Enter the value of k: ");
        int k = scanner.nextInt();

        int result = countPairs(arr, k);
        System.out.println("Number of valid pairs: " + result);
    }
}
